### ISSUE TITLE
SyntaxError in node's strict mode (--use_stict)

### ISSUE THREAD

**Issue by tsabolov** [digitalbazaar/forge#333]
Trying to require library in strict mode of node (`node --use_strict`) causes a SyntaxError:

``` javascript
let forge = require('node-forge');
```

```
SyntaxError: Delete of an unqualified identifier in strict mode.
```

The error is caused by the following line in file [/js/forge.js](https://github.com/digitalbazaar/forge/blob/master/js/forge.js#L53):

``` javascript
delete define;
```

AFAIK, in strict it's prohibited to delete global variables (properties of global object) this way. Please fix this behaviour in order to make the library work in strict mode.

**Comment by dlongley** [comment]
This is related to item 6 of the roadmap: #203

We need to update the packaging system for forge, removing handcrafted boilerplate and replacing it with something that is auto-generated and that supports a variety of different package/module loading APIs/mechanisms.

**Comment by dev0x10** [comment]
+1
Any temporary solution for this issue?

**Comment by tsabolov** [comment]
Personally I have ended up rewriting my files to have `'use strict';` in the beginning, and now I'm running node without `--use_strict` flag. This way everything is working like a charm.

**Comment by dev0x10** [comment]
I'm using webpack to build my js file which means 'use strict' will included in my bundle file.
In the end I have to do plain copy forge.min.js without compile it into my bundle js

**Comment by ysangkok** [comment]
My pull request that removes RequireJS and uses Webpack instead also runs all non-browser tests with strict mode without problems: https://github.com/digitalbazaar/forge/pull/357

**Comment by irtazabbas** [comment]
+1

**Comment by ErkanSen14** [comment]
Any permanent fix coming to this?

**Comment by dlongley** [comment]
> Any permanent fix coming to this?

cc: @davidlehn -- can we get a synopsis of the remaining challenges over at #357? There's also an offer for assistance over there.

**Comment by andreassolberg** [comment]
+1

**Comment by zloylos** [comment]
+1

**Comment by dlongley** [comment]
Should be addressed in a new release now that #456 has been merged.
